packages/sdk/src/endpoints/Endpoint.ts

```diff
@@ -47,7 +47,10 @@ export abstract class Endpoint {
 
         if (httpResponse.status !== expectedStatus) {
             const errorPayload = httpResponse.data.error;
```

Comment:
when does this happen? when you call a route that does not exist on the connector?

Reply:
oh, I didn't though about that.

Reply:
>when you use a non connect web server

You wouldn't get an HTTP response but an exception instead in that case and `makeResult` would never execute (for example, coming from a call to `Endpoint.post()`). If you want to catch socket hangups etc. you'd have to wrap the HTTP requests in a `try/catch` block and catch `AxiosError`s. `makeResult` is only ever entered after a completed HTTP request/response cycle.

Reply:
Actually it depends on how you configure axios.

Reply:
Unless I'm mistaken, axios uses exceptions to signal errors on the network level (socket hangups, unknown hosts, timeouts, ...) and this behavior cannot be changed. So if we want the SDK to gracefully handle these errors, we have to wrap the calls to `this.httpClient.{get,post, ...}()` inside _Endpoint.ts_.

What can be configured though is to not raise an exception for non-2xx and non-3xx HTTP Response codes, which it does by default (and the SDK does override this default behavior, see [here](https://github.com/nmshd/connector/blob/f3a219d13eaae672f3d188db81809f7413bcbd9d/packages/sdk/src/ConnectorClient.ts#L38)). May you're referring to this?

Reply:
Yeah, can be that I confused that. But then I am a bit confused about this PR. I don't see the usecase if network errors are already thrown errors.

Reply:
Hm, yea I rechecked @slavistan is right, real network errors are thrown. This code would only be run when running against a non connector web server or also a proxy that sends some error data back that is not in the connector format.

Reply:
But I think this is still ok.